### ISSUE TITLE
  ZWayServer ERROR! Failed to configure required characteristic "Contact Sensor State"! +1ms

### ISSUE THREAD

**Issue by aivs** [SphtKr/homebridge-zway#57]
Loading stopped at line:
`ZWayServer ERROR! Failed to configure required characteristic "Contact Sensor State"! +1ms`

I use latest Z-Way from Git and Node v6.8.1 on RPI 3 (Tested with Node 4.6, the same error).

Full Log:

```
pi@raspberrypi:~/.homebridge $ DEBUG=* homebridge
*** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs>
*** WARNING *** The program 'nodejs' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs&f=DNSServiceRegister>
[10/17/2016, 3:59:32 PM] Loaded plugin: homebridge-zway
[10/17/2016, 3:59:32 PM] Registering accessory 'homebridge-zway.ZWayServer'
[10/17/2016, 3:59:32 PM] Registering platform 'homebridge-zway.ZWayServer'
[10/17/2016, 3:59:32 PM] ---
[10/17/2016, 3:59:32 PM] Loaded config.json with 0 accessories and 1 platforms.
[10/17/2016, 3:59:32 PM] ---
[10/17/2016, 3:59:32 PM] Loading 1 platforms...
[10/17/2016, 3:59:32 PM] Initializing ZWayServer platform...
  ZWayServer Fetching Z-Way devices... +0ms
Failed to find plugin to handle accessory Датчик открытия
  ZWayServer Authenticating... +276ms
  ZWayServer Authenticated. Resubmitting original request... +128ms
  ZWayServer Got typeKey battery.Battery for vdev BatteryPolling_7 +69ms
  ZWayServer { deviceType: 'battery', probeTitle: 'Battery', probeType: '' } +1ms
  ZWayServer Got typeKey text for vdev InfoWidget_5_Int +22ms
  ZWayServer { deviceType: 'text', probeTitle: undefined, probeType: '' } +0ms
  ZWayServer Got typeKey toggleButton for vdev Notification_10 +2ms
  ZWayServer { deviceType: 'toggleButton', probeTitle: undefined, probeType: '' } +1ms
  ZWayServer Got typeKey switchBinary for vdev ZWayVDev_zway_4-0-37 +2ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: '' } +0ms
  ZWayServer Got typeKey sensorMultilevel.meterElectric_kilowatt_per_hour for vdev ZWayVDev_zway_4-0-50-0 +2ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Electric', probeType: 'meterElectric_kilowatt_per_hour' } +0ms
  ZWayServer Got typeKey sensorMultilevel.meterElectric_watt for vdev ZWayVDev_zway_4-0-50-2 +3ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Electric', probeType: 'meterElectric_watt' } +0ms
  ZWayServer Got typeKey switchBinary for vdev ZWayVDev_zway_30-0-37 +2ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: '' } +0ms
  ZWayServer Got typeKey switchMultilevel.blind for vdev ZWayVDev_zway_30-0-38 +2ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'motor' } +0ms
  ZWayServer Got typeKey sensorMultilevel.Power for vdev ZWayVDev_zway_30-0-49-4 +2ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Power', probeType: 'energy' } +0ms
  ZWayServer Got typeKey sensorMultilevel.meterElectric_kilowatt_per_hour for vdev ZWayVDev_zway_30-0-50-0 +1ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Electric', probeType: 'meterElectric_kilowatt_per_hour' } +0ms
  ZWayServer Got typeKey sensorMultilevel.meterElectric_watt for vdev ZWayVDev_zway_30-0-50-2 +2ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Electric', probeType: 'meterElectric_watt' } +0ms
  ZWayServer Got typeKey switchBinary for vdev ZWayVDev_zway_31-0-37 +1ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: '' } +1ms
  ZWayServer Got typeKey switchMultilevel for vdev ZWayVDev_zway_33-0-38 +1ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'multilevel' } +0ms
  ZWayServer Got typeKey sensorMultilevel.Temperature for vdev ZWayVDev_zway_33-0-49-1 +2ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Temperature', probeType: 'temperature' } +0ms
  ZWayServer Got typeKey switchBinary.thermostat_mode for vdev ZWayVDev_zway_33-0-64 +1ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: 'thermostat_mode' } +1ms
  ZWayServer Got typeKey thermostat for vdev ZWayVDev_zway_33-0-67-1 +1ms
  ZWayServer { deviceType: 'thermostat', probeTitle: undefined, probeType: 'thermostat_set_point' } +0ms
  ZWayServer Got typeKey battery.Battery for vdev ZWayVDev_zway_33-0-128 +2ms
  ZWayServer { deviceType: 'battery', probeTitle: 'Battery', probeType: '' } +0ms
  ZWayServer Got typeKey switchMultilevel for vdev ZWayVDev_zway_34-0-38 +1ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'multilevel' } +0ms
  ZWayServer Got typeKey switchRGBW for vdev ZWayVDev_zway_34-0-51-rgb +2ms
  ZWayServer { deviceType: 'switchRGBW', probeTitle: undefined, probeType: 'switchColor_rgb' } +0ms
  ZWayServer Got typeKey switchMultilevel for vdev ZWayVDev_zway_34-0-51-0 +1ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'switchColor_soft_white' } +0ms
  ZWayServer Got typeKey switchMultilevel for vdev ZWayVDev_zway_34-0-51-1 +4ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'switchColor_cold_white' } +0ms
  ZWayServer Got typeKey switchMultilevel for vdev ZWayVDev_zway_124-0-38 +2ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'multilevel' } +1ms
  ZWayServer Got typeKey switchMultilevel for vdev ZWayVDev_zway_124-1-38 +3ms
  ZWayServer { deviceType: 'switchMultilevel', probeTitle: undefined, probeType: 'multilevel' } +1ms
  ZWayServer Got typeKey switchBinary for vdev ZWayVDev_zway_126-0-37 +4ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: '' } +0ms
  ZWayServer Got typeKey sensorBinary.General purpose for vdev ZWayVDev_zway_129-0-48-1 +4ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'General purpose', probeType: 'general_purpose' } +1ms
  ZWayServer Got typeKey sensorBinary.Smoke for vdev ZWayVDev_zway_130-0-48-2 +2ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'Smoke', probeType: 'smoke' } +1ms
  ZWayServer Got typeKey sensorBinary.Tamper for vdev ZWayVDev_zway_130-0-48-8 +1ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'Tamper', probeType: 'tamper' } +23ms
  ZWayServer Got typeKey battery.Battery for vdev ZWayVDev_zway_130-0-128 +5ms
  ZWayServer { deviceType: 'battery', probeTitle: 'Battery', probeType: '' } +1ms
  ZWayServer Got typeKey sensorBinary.General purpose for vdev ZWayVDev_zway_132-0-48-1 +1ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'General purpose', probeType: 'general_purpose' } +1ms
  ZWayServer Got typeKey sensorMultilevel.Temperature for vdev ZWayVDev_zway_132-0-49-1 +1ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Temperature', probeType: 'temperature' } +0ms
  ZWayServer Got typeKey sensorMultilevel.Luminiscence for vdev ZWayVDev_zway_132-0-49-3 +2ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Luminiscence', probeType: 'luminosity' } +0ms
  ZWayServer Got typeKey sensorMultilevel.Humidity for vdev ZWayVDev_zway_132-0-49-5 +4ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Humidity', probeType: 'humidity' } +0ms
  ZWayServer Got typeKey sensorMultilevel.Ultraviolet for vdev ZWayVDev_zway_132-0-49-27 +2ms
  ZWayServer { deviceType: 'sensorMultilevel', probeTitle: 'Ultraviolet', probeType: 'ultraviolet' } +0ms
  ZWayServer Got typeKey sensorBinary for vdev ZWayVDev_zway_132-0-113-7-3-A +1ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: undefined, probeType: 'alarm_burglar' } +1ms
  ZWayServer Got typeKey sensorBinary for vdev ZWayVDev_zway_132-0-113-7-8-A +1ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: undefined, probeType: 'alarm_burglar' } +0ms
  ZWayServer Got typeKey battery.Battery for vdev ZWayVDev_zway_132-0-128 +1ms
  ZWayServer { deviceType: 'battery', probeTitle: 'Battery', probeType: '' } +1ms
  ZWayServer Got grouped device BatteryPolling_7 consisting of devices: +1ms
  ZWayServer BatteryPolling_7 - battery.Battery +0ms
  ZWayServer WARN: Didn't find suitable device class! +1ms
  ZWayServer Got grouped device InfoWidget_5_Int consisting of devices: +0ms
  ZWayServer InfoWidget_5_Int - text +0ms
  ZWayServer WARN: Didn't find suitable device class! +0ms
  ZWayServer Got grouped device Notification_10 consisting of devices: +1ms
  ZWayServer Notification_10 - toggleButton +0ms
  ZWayServer WARN: Didn't find suitable device class! +0ms
  ZWayServer Got grouped device ZWayVDev_4-0 consisting of devices: +0ms
  ZWayServer ZWayVDev_zway_4-0-37 - switchBinary +0ms
  ZWayServer ZWayVDev_zway_4-0-50-0 - sensorMultilevel.Electric +0ms
  ZWayServer ZWayVDev_zway_4-0-50-2 - sensorMultilevel.Electric +1ms
  ZWayServer Got grouped device ZWayVDev_30-0 consisting of devices: +1ms
  ZWayServer ZWayVDev_zway_30-0-37 - switchBinary +0ms
  ZWayServer ZWayVDev_zway_30-0-38 - switchMultilevel +0ms
  ZWayServer ZWayVDev_zway_30-0-49-4 - sensorMultilevel.Power +0ms
  ZWayServer ZWayVDev_zway_30-0-50-0 - sensorMultilevel.Electric +0ms
  ZWayServer ZWayVDev_zway_30-0-50-2 - sensorMultilevel.Electric +1ms
  ZWayServer Got grouped device ZWayVDev_31-0 consisting of devices: +0ms
  ZWayServer ZWayVDev_zway_31-0-37 - switchBinary +0ms
  ZWayServer Got grouped device ZWayVDev_33-0 consisting of devices: +1ms
  ZWayServer ZWayVDev_zway_33-0-38 - switchMultilevel +0ms
  ZWayServer ZWayVDev_zway_33-0-49-1 - sensorMultilevel.Temperature +0ms
  ZWayServer ZWayVDev_zway_33-0-64 - switchBinary +0ms
  ZWayServer ZWayVDev_zway_33-0-67-1 - thermostat +0ms
  ZWayServer ZWayVDev_zway_33-0-128 - battery.Battery +0ms
  ZWayServer Got grouped device ZWayVDev_34-0 consisting of devices: +1ms
  ZWayServer ZWayVDev_zway_34-0-38 - switchMultilevel +0ms
  ZWayServer ZWayVDev_zway_34-0-51-rgb - switchRGBW +0ms
  ZWayServer ZWayVDev_zway_34-0-51-0 - switchMultilevel +0ms
  ZWayServer ZWayVDev_zway_34-0-51-1 - switchMultilevel +0ms
  ZWayServer Got grouped device ZWayVDev_124-0 consisting of devices: +4ms
  ZWayServer ZWayVDev_zway_124-0-38 - switchMultilevel +1ms
  ZWayServer Got grouped device ZWayVDev_124-1 consisting of devices: +0ms
  ZWayServer ZWayVDev_zway_124-1-38 - switchMultilevel +0ms
  ZWayServer Got grouped device ZWayVDev_126-0 consisting of devices: +0ms
  ZWayServer ZWayVDev_zway_126-0-37 - switchBinary +0ms
  ZWayServer Got grouped device ZWayVDev_129-0 consisting of devices: +1ms
  ZWayServer ZWayVDev_zway_129-0-48-1 - sensorBinary.General purpose +0ms
  ZWayServer Got grouped device ZWayVDev_130-0 consisting of devices: +0ms
  ZWayServer ZWayVDev_zway_130-0-48-2 - sensorBinary.Smoke +0ms
  ZWayServer ZWayVDev_zway_130-0-48-8 - sensorBinary.Tamper +1ms
  ZWayServer ZWayVDev_zway_130-0-128 - battery.Battery +0ms
  ZWayServer WARN: Didn't find suitable device class! +2ms
  ZWayServer Got grouped device ZWayVDev_132-0 consisting of devices: +3ms
  ZWayServer ZWayVDev_zway_132-0-48-1 - sensorBinary.General purpose +1ms
  ZWayServer ZWayVDev_zway_132-0-49-1 - sensorMultilevel.Temperature +1ms
  ZWayServer ZWayVDev_zway_132-0-49-3 - sensorMultilevel.Luminiscence +1ms
  ZWayServer ZWayVDev_zway_132-0-49-5 - sensorMultilevel.Humidity +0ms
  ZWayServer ZWayVDev_zway_132-0-49-27 - sensorMultilevel.Ultraviolet +0ms
  ZWayServer ZWayVDev_zway_132-0-113-7-3-A - sensorBinary +0ms
  ZWayServer ZWayVDev_zway_132-0-113-7-8-A - sensorBinary +0ms
  ZWayServer ZWayVDev_zway_132-0-128 - battery.Battery +0ms
....
[10/17/2016, 3:59:33 PM] Initializing platform accessory 'Z-Wave.Me Switch (126.0)'...
  ZWayServer Got typeKey switchBinary for vdev ZWayVDev_zway_126-0-37 +10ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: '' } +1ms
  ZWayServer Got typeKey switchBinary for vdev ZWayVDev_zway_126-0-37 +3ms
  ZWayServer { deviceType: 'switchBinary', probeTitle: undefined, probeType: '' } +1ms
  ZWayServer Loaded services for Z-Wave.Me Switch (126.0) +1ms
[10/17/2016, 3:59:33 PM] Initializing platform accessory 'Z-Wave.Me General purpose (129.0.48.1)'...
  ZWayServer Got typeKey sensorBinary.General purpose for vdev ZWayVDev_zway_129-0-48-1 +9ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'General purpose', probeType: 'general_purpose' } +0ms
  ZWayServer Got typeKey sensorBinary.General purpose for vdev ZWayVDev_zway_129-0-48-1 +10ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'General purpose', probeType: 'general_purpose' } +0ms
  ZWayServer Got typeKey sensorBinary.General purpose for vdev ZWayVDev_zway_129-0-48-1 +1ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'General purpose', probeType: 'general_purpose' } +0ms
  ZWayServer Got typeKey sensorBinary.General purpose for vdev ZWayVDev_zway_129-0-48-1 +1ms
  ZWayServer { deviceType: 'sensorBinary', probeTitle: 'General purpose', probeType: 'general_purpose' } +0ms
  ZWayServer ERROR! Failed to configure required characteristic "Contact Sensor State"! +2ms
```

**Comment by aivs** [comment]
I deactivated device_129 and now its working. device_129 from config.json:
```
{
"creationTime": 1476443139,
"creatorId": 1,
"deviceType": "sensorBinary",
"h": 748828025,
"hasHistory": false,
"id": "ZWayVDev_zway_129-0-48-1",
"location": 0,
"metrics": {
"probeTitle": "General purpose",
"scaleTitle": "",
"icon": "motion",
"level": "off",
"title": "Z-Wave.Me General purpose (129.0.48.1)"
},
"permanently_hidden": true,
"probeType": "general_purpose",
"tags": [],
"visibility": true,
"updateTime": 1476714704
},
```

**Comment by klaasjoerg** [comment]
I had a similar problem, but solved it differently, by changing the index.js .... see #56

**Comment by SphtKr** [comment]
So, I can see it failing to configure the device, but it shouldn't quit in that case, it should just skip to the next one. So while part of your issue is the same as #53, there's something else going on there. 

Does device 129 just happen to be your last device or are there more?

**Comment by SphtKr** [comment]
Oh...and what exactly _is_ device 129? Manufacturer/model?

**Comment by klaasjoerg** [comment]
In my case: #56:
a.) It wasn't the last device. It just happened in the middle of something.
b.) In my case it was the Fibaro Door/Window contact sensor if I remember correctly...

**Comment by SphtKr** [comment]
Believed fixed in 9d163e6, reopen or create a new issue if not fixed.
